### ISSUE TITLE
Performance dropped to zero.

### ISSUE THREAD

**Issue by icambridge** [src-d/go-git#254]
Firstly, want to thank you guys for the work on this library. 

So I upgraded to the latest and greatest. And my performance dropped to zero. On 8 core, 16 GB, SSD, processor going at 700+%. Zero repositories were processed in 10 minutes.

I've been trying to revert to older stuff or newer stuff but because of BC changes made in child libs (go-billy) I can't. I can't get the actual latest and greatest because of the way you guys are handling release management.

Doing debugging locally on a Macbook Pro. My app hangs on cloning 100MB worth of repositories. which were all downloaded. So it looks like you introduced a killer performance issue in to your filesystem handling.

I'm also getting this if I use memory storage or file system storage.

So to recap.

* Release management is painful. You've tagged stuff on github but it doesn't come through via your vanity url. Looks like you're blocking the new `dep` tools functionalities. (Maybe I should create a new ticket for this?)
* Performance is non existent.
* Both filesystem and memory storage hang on clone.

**Comment by smola** [comment]
@icambridge Thanks for the report.

* With respect to the import URL: @mcuadros maybe you know what's going on with the import URL here?
* Indeed, there was compatibility breakage at go-billy, which shouldn't happen again (until v2, of course).
* Performance problems should be solved soon. There has been some bugs introduced recently (actually while doing some performance optimizations).

Please, whenever you have multiple problems, try to file a different issue for each one. That'll allow us to handle them more nicely (closing them independently, etc).
